webapp/tests/model/test_eligibility_data.py

```diff
@@ -1038,41 +1021,72 @@ def test_if_marginal_employ_valid_and_other_income_no_then_raise_no_validation_e
         try:
             with patch('app.model.eligibility_data.OtherIncomeEligibilityData.__init__',
-                       MagicMock(return_value=None)):
+                        MagicMock(return_value=None)):
                 ForeignCountrySuccessEligibility.parse_obj(valid_data)
         except ValidationError:
             self.fail("ForeignCountryEligibility.parse_obj should not raise validation error")
 
+    def test_if_other_income_valid_and_foreign_country_no_and_both_persons_defined_and_both_person_has_elster_account_eligibility_then_raise_validation_error(self):
+        non_valid_data = {'foreign_country_eligibility': 'no', 'user_a_has_elster_account_eligibility': 'yes', 'user_b_has_elster_account_eligibility': 'yes'}
+        with patch('app.model.eligibility_data.OtherIncomeEligibilityData.__init__', MagicMock(return_value=None)):
+            self.assertRaises(ValidationError, ForeignCountrySuccessEligibility.parse_obj, non_valid_data)
+
+    def test_if_other_income_valid_and_foreign_country_yes_then_raise_validation_error(self):
+        non_valid_data = {'foreign_country_eligibility': 'yes'}
+        with patch('app.model.eligibility_data.OtherIncomeEligibilityData.__init__', MagicMock(return_value=None)):
+            self.assertRaises(ValidationError, ForeignCountrySuccessEligibility.parse_obj, non_valid_data)
+
+    def test_if_other_income_invalid_and_foreign_country_no_then_raise_validation_error(self):
+        valid_data = {'foreign_country_eligibility': 'no'}
+        with patch('app.model.eligibility_data.OtherIncomeEligibilityData.__init__', 
+                   MagicMock(side_effect=ValidationError([], OtherIncomeEligibilityData))):
+            self.assertRaises(ValidationError, ForeignCountrySuccessEligibility.parse_obj, valid_data)
+
+    def test_if_other_income_valid_and_foreign_country_no_and_user_a_has_elster_account_eligibility_then_raise_validation_error(self):
+        valid_data = {'foreign_country_eligibility': 'no', 'user_a_has_elster_account_eligibility': 'yes'}
+        with patch('app.model.eligibility_data.OtherIncomeEligibilityData.__init__', 
+                   MagicMock(side_effect=ValidationError([], OtherIncomeEligibilityData))):
+            self.assertRaises(ValidationError, ForeignCountrySuccessEligibility.parse_obj, valid_data)
+            
+    def test_if_other_income_valid_and_foreign_country_no_and_user_a_has_no_elster_account_eligibility_then_raise_no_validation_error(self):
+        valid_data = {'foreign_country_eligibility': 'no', 'user_a_has_elster_account_eligibility': 'no'}
+        with patch('app.model.eligibility_data.OtherIncomeEligibilityData.__init__', 
+                   MagicMock(side_effect=ValidationError([], OtherIncomeEligibilityData))):
+            self.assertRaises(ValidationError, ForeignCountrySuccessEligibility.parse_obj, valid_data)
```

Comment:
I think this is a duplication to l. 1025?

Reply:
yes :see_no_evil: